src/lib/github/issueCommentEditAction.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * function that enables an issue comment to be editable, that is to say,
+ * provides the delete and edit(overwrite) options.
+ *
+ * @param: context & issueCommentArray
+ * @returns: Promise<void>
+ */
+
+import { getBasicRepoProps } from "./utils";
+import { Context } from "./types";
+
+type CommentEditAction = "Delete" | "Edit";
+
+const Delete: CommentEditAction = "Delete";
+const Edit: CommentEditAction = "Edit";
+
+interface CommentEdit {
+  GhIssueId: number;
+  GhCommentId: number;
+  action: CommentEditAction;
+  commentsText: string[];
+}
+
+type CommentsEdit = CommentEdit[];
+
+export default async function issueCommentToEdit (context: Context, issueComment: CommentsEdit): Promise<void> {
+    const octokit = context.github;
+    const { owner, repo } = getBasicRepoProps(context);
+    issueComment.map(async (comment) => {
+        const commentId = await octokit.issues.getComment({owner, repo, comment_id: comment.GhCommentId});
+        if (comment.action === Delete) {
+            return await octokit.issues.deleteComment({owner, repo, comment_id: commentId});
+        }
+        if (comment.action === Edit) {
+            return await octokit.issues.updateComment({owner, repo, comment_id: commentId, body: comment.commentsText});
```

Comment:
You can also make this shorter this way
```
 if (comment.action === Edit) {
            return await octokit
                .issues
                .updateComment({
                    owner,
                    repo,
                    comment_id: commentId,
                    body: comment.commentsText
                });
        }
```
Secondly `body` in the `updateComment` function should be a string not an `array`. Currently comment.commentsText is an array. In the interface for `CommentEdit` turn it into a string.